tests/components/gios/test_air_quality.py

```diff
@@ -0,0 +1,123 @@
+"""Test air_quality of GIOS integration."""
+from datetime import timedelta
+import json
+
+from gios import ApiError
+
+from homeassistant.components.air_quality import (
+    ATTR_AQI,
+    ATTR_CO,
+    ATTR_NO2,
+    ATTR_OZONE,
+    ATTR_PM_2_5,
+    ATTR_PM_10,
+    ATTR_SO2,
+)
+from homeassistant.components.gios.air_quality import ATTRIBUTION
+from homeassistant.components.gios.const import AQI_GOOD
+from homeassistant.const import (
+    ATTR_ATTRIBUTION,
+    ATTR_ICON,
+    ATTR_UNIT_OF_MEASUREMENT,
+    CONCENTRATION_MICROGRAMS_PER_CUBIC_METER,
+    STATE_UNAVAILABLE,
+)
+from homeassistant.util.dt import utcnow
+
+from tests.async_mock import patch
+from tests.common import async_fire_time_changed, load_fixture
+from tests.components.gios import init_integration
+
+
+async def test_air_quality(hass):
+    """Test states of the air_quality."""
+    await init_integration(hass)
+    registry = await hass.helpers.entity_registry.async_get_registry()
+
+    state = hass.states.get("air_quality.home")
+    assert state
+    assert state.state == "4"
+    assert state.attributes.get(ATTR_ATTRIBUTION) == ATTRIBUTION
+    assert state.attributes.get(ATTR_AQI) == AQI_GOOD
+    assert state.attributes.get(ATTR_PM_10) == 17
+    assert state.attributes.get(ATTR_PM_2_5) == 4
+    assert state.attributes.get(ATTR_CO) == 252
+    assert state.attributes.get(ATTR_SO2) == 4
```

Comment:
Codacy found an issue: [Use of assert detected. The enclosed code will be removed when compiling to optimised byte code.](https://app.codacy.com/manual/foreign-sub/home-assistant/pullRequest?prid=6103615)